src/ch/ch_driver.c

```diff
@@ -1318,6 +1319,289 @@ chDomainPinVcpu(virDomainPtr dom,
                                   VIR_DOMAIN_AFFECT_LIVE);
 }
 
+#define CH_NB_NUMA_PARAM 2
+
+static int
+chDomainGetNumaParameters(virDomainPtr dom,
+                          virTypedParameterPtr params,
+                          int *nparams,
+                          unsigned int flags)
+{
+    size_t i;
+    virDomainObjPtr vm = NULL;
+    virDomainNumatuneMemMode tmpmode = VIR_DOMAIN_NUMATUNE_MEM_STRICT;
+    virCHDomainObjPrivatePtr priv;
+    g_autofree char *nodeset = NULL;
+    int ret = -1;
+    virDomainDefPtr def = NULL;
+    bool live = false;
+    virBitmapPtr autoNodeset = NULL;
+
+    virCheckFlags(VIR_DOMAIN_AFFECT_LIVE |
+                  VIR_DOMAIN_AFFECT_CONFIG |
+                  VIR_TYPED_PARAM_STRING_OKAY, -1);
+
+    VIR_INFO("virCHDomainObjFromDomain..");
+    if (!(vm = virCHDomainObjFromDomain(dom)))
+        return -1;
+    priv = vm->privateData;
+
+    VIR_INFO("virDomainGetNumaParametersEnsureACL..");
+    if (virDomainGetNumaParametersEnsureACL(dom->conn, vm->def) < 0)
+        goto cleanup;
+
+    VIR_INFO("virDomainObjGetOneDefState");
+    if (!(def = virDomainObjGetOneDefState(vm, flags, &live)))
+        goto cleanup;
+
+    if (live)
+        autoNodeset = priv->autoNodeset;
+
+    if ((*nparams) == 0) {
+        *nparams = CH_NB_NUMA_PARAM;
+        ret = 0;
+        goto cleanup;
+    }
+
+    for (i = 0; i < CH_NB_NUMA_PARAM && i < *nparams; i++) {
+        virMemoryParameterPtr param = &params[i];
+
+        switch (i) {
+        case 0: /* fill numa mode here */
+            ignore_value(virDomainNumatuneGetMode(def->numa, -1, &tmpmode));
+
+            VIR_INFO("virTypedParameterAssign after virDomainNumatuneGetMode");
```

Comment:
Just a blanket comment for the rest of the VIR_INFO prints. If these are important, please keep but largely they look like debug prints that can be cleaned up.

Reply:
I cleaned up all unnecessary VIR_INFO prints